src/main/java/com/ita/if103java/ims/controller/UpgradeController.java

```diff
@@ -21,6 +25,21 @@ public UpgradeController(UpgradeService upgradeService) {
 
     @PutMapping("/{typeId}")
     public void upgrade(@AuthenticationPrincipal User user, @PathVariable("typeId") Long typeId) {
-        upgradeService.upgradeAccount(user.getAccountId(), typeId);
+        upgradeService.upgradeAccount(user, typeId);
+    }
+
+    @GetMapping(value = "/")
+    public AccountTypeDto findCurrentType(@AuthenticationPrincipal User user) {
```

Comment:
Just noticed that you use user, but as far as i remember Maria created UserDetailsImpl class which will represent currently logged in user. @MarrMika , correct?
@Mychalevytsh @BohdanAndrusiak @rostIvan @Hvozdianchuck @volodymyrvynnyk @pustoj0 
please pay attention on this, maybe you need to update your classes too.

Reply:
It is correct. You are right!